modules/lang-painless/src/main/java/org/elasticsearch/painless/PainlessExecuteAction.java

```diff
@@ -135,33 +348,45 @@ public ActionRequestValidationException validate() {
             if (script.getType() != ScriptType.INLINE) {
                 validationException = addValidationError("only inline scripts are supported", validationException);
             }
+            if (needDocumentAndIndex(executeScriptContext.scriptContext)) {
+                if (executeScriptContext.index == null) {
+                    validationException = addValidationError("index is a required parameter for current context", validationException);
+                }
+                if (executeScriptContext.document == null) {
+                    validationException = addValidationError("document is a required parameter for current context", validationException);
+                }
+            }
             return validationException;
         }
 
         @Override
         public void readFrom(StreamInput in) throws IOException {
             super.readFrom(in);
             script = new Script(in);
-            context = SupportedContext.fromId(in.readByte());
+            if (in.getVersion().onOrBefore(Version.V_6_4_0)) {
```

Comment:
Isn't the execute action only in 6.x right now? Why would wire compat be necessary when no release includes it yet?

Reply:
It is also in the 6.3 branch. Unless we want to get this change into 6.3 branch too (while 6.3.0 has not been released) then I think we should have this wire backwards compatibility logic.

Reply:
This API is marked as experimental, which allows us to make changes to request / response format. I don't know what we exactly guarantee for experimental APIs in a cluster with mixed nodes, but if we don't have any guarantees for that then the wire backwards compatibility logic can be removed.